### ISSUE TITLE
how to avoid filename truncation in listings?

### ISSUE THREAD

**Issue by kberry** [viewvc/viewvc#332]
In viewvc-1.3.0-dev, at least as of 20230429, filenames are truncated in the display at some relatively narrow width, with an ellipsis following. Screenshot attached for https://tug.org/svn/tugboat/trunk/covers/. I guess this is being in the css, since the entire filename (e.g., "tubtocmac.tex") is present in the source.

Unfortunately my css-fu is minimal, and I've been unable to discern where this is happening and, more importantly, what needs to be changed to either increase the maximum width or (ideally) turn off such truncation completely.

Help? Any info greatly appreciated. Thanks.

**Comment by kberry** [comment]
sorry, i can't figure out how to attach a screenshot to the issue, but hopefully the url is enough.

**Comment by futatuki** [comment]
I've seen with Firefox on FreeBSD, but I can't reproduce it.  Even with minimum width 450 pixel and max  render size 500% it shows full file name, although last change field disapeared.

![](file:///home/futatuki/tmp/viewvc-sc-issue-332.jpg)

**Comment by futatuki** [comment]
To attach a file in comments by using web browzer,  it needs to allow to use all scripts for the page, and then click "Attach files by dragging & dropping, selecting or pasting them."  field and select file to upload.

**Comment by futatuki** [comment]
In default template,  the width of the file name field in directory view is specified in `templates/default/directory.ezt`,  

`<th style="width: 200px" class="vc_header[is sortby "file"]_sort[end]">`
and
`<td style="width: 200px" [if-any click_href]onclick="jumpTo('[click_href]')"
[end]>`

lines. 

So if  you make a custom template from default template, by removing '`style="width:200px"`' the field width can be adjusted automatically,  or  increasing amount of width, it could ease the problem,  I guess.

**Comment by kberry** [comment]
Thanks for all the replies, including pointing out my blindness to
adding github attachments.

    I've seen with Firefox on FreeBSD, but I can't reproduce it. 

It seems bizarre that this is browser-dependent, let alone os-dependent
-- that's pretty basic css. But I guess it is.

For the record, I'm currently seeing this with Firefox 102.15.1esr on
Alma Linux.  Also previously with other Firefox versions and, I'm pretty
sure, with Chrome.

    In default template, the width of the file name field in directory view
    is specified in `templates/default/directory.ezt`,

I tried changing that file, and it worked -- once I realized that either
firefox or viewvc was somehow not remaking the html even when I hit
"reload" in the browser, or started again in a new tab, etc. That's a
level of caching I haven't seen before with cgi's. But going to a
different directory showed the changes. Yay! Thank you!!

BTW, I know you said "make a custom template", but unfortunately I've
never succeeded in doing that. Nothing ever seems to activate the new
template, i.e., I set up the new directory, copy files, etc., but
changing template_dir in viewvc.conf had no evident effect. But maybe
that was also due to the caching, and I didn't realize what I was
(not) testing.

(Aside: I think I will try to add explicit a "no caching" <meta> to
_header.ezt and see if that helps.)

Thanks again,
Karl
